Tennis-scheduler/src/main/java/com/tennis/tennisscheduler/controllers/TimeslotController.java

```diff
@@ -53,23 +66,47 @@ public ResponseEntity<TimeslotDto> getById(@PathVariable long id){
+
         TimeslotResponseDto timeslotResponse = new TimeslotResponseDto();
-        if (!result.hasErrors()){
+        if (!result.hasErrors() && user.getRole().getRoleName().equals(UserType.ROLE_TENNIS_PLAYER)){
+            timeslotNew.personId = user.getId();
             timeslotResponse = timeslotResponseDtoMapper.toTimeslotResponseDto(timeslotService.reserveTimeslot(timeslotDtoMapper.fromTimeslotDtoToTimeslot(timeslotNew)));
             return new ResponseEntity<>(timeslotResponse, HttpStatus.CREATED);
         }
-        else {
+        else if(!result.hasErrors() && user.getRole().getRoleName().equals(UserType.ROLE_ADMIN)){
+            timeslotResponse = timeslotResponseDtoMapper.toTimeslotResponseDto(timeslotService.reserveTimeslot(timeslotDtoMapper.fromTimeslotDtoToTimeslot(timeslotNew)));
+            return new ResponseEntity<>(timeslotResponse, HttpStatus.CREATED);
+        } else {
             timeslotResponse.message = result.getAllErrors();
             return new ResponseEntity<>(timeslotResponse, HttpStatus.BAD_REQUEST);
         }
     }
 
     @PreAuthorize("hasAnyRole('ADMIN','TENNIS_PLAYER')")
     @PutMapping(value = "/{id}")
-    public ResponseEntity<TimeslotResponseDto> update(@PathVariable long id, @RequestBody TimeslotDto timeslotUpdate){
+    public ResponseEntity<TimeslotResponseDto> update(@PathVariable long id, @RequestBody @Valid TimeslotDto timeslotUpdate){
+        Authentication authentication = SecurityContextHolder.getContext().getAuthentication();
+        Person user = (Person)authentication.getPrincipal();
+
         Timeslot timeslotExisting = timeslotService.getById(id);
-        if (timeslotExisting == null) {
+        if (timeslotExisting == null)
             return new ResponseEntity<>( HttpStatus.NOT_FOUND);
+
+        if(user.getRole().getRoleName().equals(UserType.ROLE_TENNIS_PLAYER)){
+            if(timeslotExisting.getPerson().getId()== user.getId()){
+                if (timeslotService.checkIfUserIsOwnerOfTimeslot(user.getId(), timeslotUpdate.id)){
+                    timeslotUpdate.personId = user.getId();
+                    TimeslotResponseDto timeslotResponse = timeslotResponseDtoMapper.toTimeslotResponseDto(timeslotService.update(id, timeslotDtoMapper.fromTimeslotDtoToTimeslot(timeslotUpdate)));
+                    if(timeslotResponse.timeslot != null)
+                        return new ResponseEntity<>(timeslotResponse, HttpStatus.OK);
+                    else
+                        return new ResponseEntity<>(timeslotResponse, HttpStatus.BAD_REQUEST);
+                } else
+                    return new ResponseEntity<>(null, HttpStatus.UNAUTHORIZED);
+            } else
+                return new ResponseEntity<>(null, HttpStatus.UNAUTHORIZED);
         }
 
         TimeslotResponseDto timeslotResponse = timeslotResponseDtoMapper.toTimeslotResponseDto(timeslotService.update(id, timeslotDtoMapper.fromTimeslotDtoToTimeslot(timeslotUpdate)));
```

Comment:
What is the actual difference between this and line 101? Why do we need it twice?